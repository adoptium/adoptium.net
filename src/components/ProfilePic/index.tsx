import React from "react"
import Image from "next/image"

interface ProfilePicProps {
  identifier: string
  name: string
}

const ProfilePic = (props: ProfilePicProps) => {
  return (
    <Image
      src={`/images/authors/${props.identifier}.jpg`}
      alt={props.name}
      width={50}
      height={50}
      className="ml-2 mb-0 rounded-full"
    />
  )
}

export default ProfilePic
