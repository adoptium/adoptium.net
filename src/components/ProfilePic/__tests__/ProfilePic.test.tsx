import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import ProfilePic from "../index"

describe("ProfilePic component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <ProfilePic identifier="pmc" name="sample name" />,
    )

    expect(container).toMatchSnapshot()
  })

  it("should render correctly - no profile pic found", () => {
    const { container } = render(<ProfilePic identifier="foo" name="Joe Bloggs" />)
    expect(container).toMatchSnapshot()
  })
})
