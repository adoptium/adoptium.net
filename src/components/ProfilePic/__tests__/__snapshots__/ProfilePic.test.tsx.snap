// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfilePic component > should render correctly - no profile pic found 1`] = `
<div>
  <img
    alt="Joe Bloggs"
    class="ml-2 mb-0 rounded-full"
    data-nimg="1"
    decoding="async"
    height="50"
    loading="lazy"
    src="/_next/image?url=%2Fimages%2Fauthors%2Ffoo.jpg&w=128&q=75"
    srcset="/_next/image?url=%2Fimages%2Fauthors%2Ffoo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fauthors%2Ffoo.jpg&w=128&q=75 2x"
    style="color: transparent;"
    width="50"
  />
</div>
`;

exports[`ProfilePic component > should render correctly 1`] = `
<div>
  <img
    alt="sample name"
    class="ml-2 mb-0 rounded-full"
    data-nimg="1"
    decoding="async"
    height="50"
    loading="lazy"
    src="/_next/image?url=%2Fimages%2Fauthors%2Fpmc.jpg&w=128&q=75"
    srcset="/_next/image?url=%2Fimages%2Fauthors%2Fpmc.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fauthors%2Fpmc.jpg&w=128&q=75 2x"
    style="color: transparent;"
    width="50"
  />
</div>
`;
